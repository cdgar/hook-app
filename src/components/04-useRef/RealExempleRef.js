import React, { useState } from 'react';
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks';
import '../02-useEffect/effects.css';

const RealExempleRef = () => {

    const [show, setShow] = useState(false)

    return (
        <div>
            <h1>RealExempleRef</h1>
            <hr/>
            {show && <MultipleCustomHooks/>}
            <button
                className="btn btn-primary mt-5"
                onClick={() => setShow(!show)}
            >
                SHow/Hide
            </button>
        </div>
    )
}

export default RealExempleRef;
