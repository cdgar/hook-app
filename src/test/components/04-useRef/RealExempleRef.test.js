import React from 'react';
import { shallow } from 'enzyme';
import RealExempleRef from '../../../components/04-useRef/RealExempleRef';

describe('RealExempleRef.js', () => {
    const wrapper = shallow(<RealExempleRef />);

    test('should show RealExempleRef', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    })

    test('should show RealExempleRef', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    })
    
})