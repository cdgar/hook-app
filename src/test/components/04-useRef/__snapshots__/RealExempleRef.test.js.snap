// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RealExempleRef.js should show RealExempleRef 1`] = `
<div>
  <h1>
    RealExempleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    SHow/Hide
  </button>
</div>
`;
